aptos-move/e2e-tests/src/loader/mod.rs

```diff
@@ -5,27 +5,44 @@
 //! Logic for account universes. This is not in the parent module to enforce privacy.
 
 use crate::{account::AccountData, executor::FakeExecutor};
+use aptos_cached_packages::aptos_stdlib;
+use aptos_framework::{BuildOptions, BuiltPackage};
 use aptos_proptest_helpers::Index;
-use move_core_types::{identifier::Identifier, language_storage::ModuleId};
-use move_ir_compiler::Compiler;
-use petgraph::{algo::toposort, Direction, Graph};
+use aptos_temppath::TempPath;
+use aptos_types::transaction::{
+    EntryFunction, ExecutionStatus, SignedTransaction, TransactionStatus,
+};
+use move_core_types::{identifier::Identifier, language_storage::ModuleId, value::MoveValue};
+use petgraph::{algo::toposort, graph::NodeIndex, Direction, Graph};
 use proptest::{
     collection::{vec, SizeRange},
     prelude::*,
 };
-use std::{cmp::Ordering, collections::HashMap};
-
+use std::cmp::Ordering;
 mod module_generator;
 
+const DEFAULT_BALANCE: u64 = 1_000_000_000;
+
 #[derive(Debug)]
 pub struct Node {
     name: ModuleId,
     self_value: u64,
     account_data: AccountData,
+    expected_value: u64,
 }
 
 #[derive(Debug)]
-pub struct DependencyGraph(Graph<Node, ()>);
+pub struct DependencyGraph {
+    graph: Graph<Node, ()>,
+    base_directory: TempPath,
+    sender_account: AccountData,
+}
+
+#[derive(Debug)]
+pub enum LoaderTransactionGen {
+    UpdateEdge(Index, Index),
+    Invoke(Index),
+}
```

Comment:
Would be helpful if you add some doc comments here explaining what these data structures represent